packages/markdown-docx/src/OOXMLTransformer/index.js

```diff
@@ -58,171 +87,178 @@ class OoxmlTransformer {
     }
 
     /**
-     * Gets the value of the variable
+     * Constructs a ciceroMark Node JSON from the information
      *
-     * @param {Array} elements the variable elements
-     * @returns {string} the value of the variable
+     * @param {object} nodeInformation Contains properties and value of a node
+     * @return {object} CiceroMark Node
      */
-    getValue(elements) {
-        const variableContents = elements[1];
-        for (const property of variableContents.elements[0].elements) {
-            if (property.name === 'w:t') {
-                return property.elements[0].text;
-            }
+    construtCiceroMarkNodeJSON(nodeInformation) {
+        let obj = {};
+        if (nodeInformation.nodeType === 'softbreak') {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Softbreak`,
+            };
+        } else if (nodeInformation.nodeType === 'variable') {
+            obj = {
+                $class: `${NS_PREFIX_CiceroMarkModel}Variable`,
+                value: nodeInformation.value || 'Not provided',
+                elementType: nodeInformation.elementType || 'Not a template variable',
+                name: nodeInformation.name || 'No name available',
+            };
+        } else {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Text`,
+                text: nodeInformation.value,
+            };
         }
-        return null;
+        for (let i = nodeInformation.properties.length - 1; i >= 0; i--) {
+            obj = {
+                $class: nodeInformation.properties[i],
+                nodes: [obj],
+            };
+        }
+        return obj;
     }
 
     /**
-     * Checks if the element is a line break or not
+     * Creates a pargraph, heading or block CiceroMark Node
      *
-     * @param {object} element the element to be checked
-     * @returns {boolean} whether the element is a line break
+     * @param {object} rootNode Pargraph, heading or any other node
      */
-    isLineBreak(element) {
-        return element.name === 'w:p' && element.elements === undefined;
-    }
+    constructNodes(rootNode) {
+        if (this.JSONXML.length > 0) {
+            let constructedNode;
+            constructedNode = this.construtCiceroMarkNodeJSON(this.JSONXML[0]);
+            rootNode.nodes = [...rootNode.nodes, constructedNode];
```

Comment:
I am assuming you're getting `rootNode` directly from the OOXML (converted to JSON by `xml-js`). Don't modify it because we don't want any sideffects.

Reply:
The `rootNode` here is the argument passed and not the one we get from a document which is converted via `xml-js`. The rootNode here is for paragraph heading or any other block node